Issue: There is one racy test

### Is there an existing issue for this?

- [X] I have searched the existing issues and found none that matched mine

### Describe the issue

Running all the tests with:

`go test ./...`, I got once:

```sh
--- FAIL: TestNew (0.00s)
    --- FAIL: TestNew/Clean_up_on_interval (2.01s)
        db_test.go:132: 
                Error Trace:    /home/didrocks/work/authd/internal/cache/db_test.go:132
                Error:          Not equal: 
                                expected: "GroupByID:\n    \"33333\": '{\"Name\":\"group3\",\"GID\":33333}'\n    \"99999\": '{\"Name\":\"commongroup\",\"GID\":99999}'\nGroupByName:\n    commongroup: '{\"Name\":\"commongroup\",\"GID\":99999}'\n    group3: '{\"Name\":\"group3\",\"GID\":33333}'\nGroupToUsers:\n    \"33333\": '{\"GID\":33333,\"UIDs\":[3333]}'\n    \"99999\": '{\"GID\":99999,\"UIDs\":[3333]}'\nUserByID:\n    \"3333\": '{\"Name\":\"user3\",\"UID\":3333,\"GID\":33333,\"Gecos\":\"User3\",\"Dir\":\"/home/user3\",\"Shell\":\"/bin/zsh\",\"LastPwdChange\":-1,\"MaxPwdAge\":-1,\"PwdWarnPeriod\":-1,\"PwdInactivity\":-1,\"MinPwdAge\":-1,\"ExpirationDate\":-1,\"LastLogin\":\"ABCDETIME\"}'\nUserByName:\n    user3: '{\"Name\":\"user3\",\"UID\":3333,\"GID\":33333,\"Gecos\":\"User3\",\"Dir\":\"/home/user3\",\"Shell\":\"/bin/zsh\",\"LastPwdChange\":-1,\"MaxPwdAge\":-1,\"PwdWarnPeriod\":-1,\"PwdInactivity\":-1,\"MinPwdAge\":-1,\"ExpirationDate\":-1,\"LastLogin\":\"ABCDETIME\"}'\nUserToGroups:\n    \"3333\": '{\"UID\":3333,\"GIDs\":[33333,99999]}'\n"
                                actual  : "GroupByID:\n    \"11111\": '{\"Name\":\"group1\",\"GID\":11111}'\n    \"22222\": '{\"Name\":\"group2\",\"GID\":22222}'\n    \"33333\": '{\"Name\":\"group3\",\"GID\":33333}'\n    \"99999\": '{\"Name\":\"commongroup\",\"GID\":99999}'\nGroupByName:\n    commongroup: '{\"Name\":\"commongroup\",\"GID\":99999}'\n    group1: '{\"Name\":\"group1\",\"GID\":11111}'\n    group2: '{\"Name\":\"group2\",\"GID\":22222}'\n    group3: '{\"Name\":\"group3\",\"GID\":33333}'\nGroupToUsers:\n    \"11111\": '{\"GID\":11111,\"UIDs\":[1111]}'\n    \"22222\": '{\"GID\":22222,\"UIDs\":[2222]}'\n    \"33333\": '{\"GID\":33333,\"UIDs\":[3333]}'\n    \"99999\": '{\"GID\":99999,\"UIDs\":[2222,3333]}'\nUserByID:\n    \"1111\": '{\"Name\":\"user1\",\"UID\":1111,\"GID\":11111,\"Gecos\":\"User1 gecos\\nOn multiple lines\",\"Dir\":\"/home/user1\",\"Shell\":\"/bin/bash\",\"LastPwdChange\":-1,\"MaxPwdAge\":-1,\"PwdWarnPeriod\":-1,\"PwdInactivity\":-1,\"MinPwdAge\":-1,\"ExpirationDate\":-1,\"LastLogin\":\"AAAAATIME\"}'\n    \"2222\": '{\"Name\":\"user2\",\"UID\":2222,\"GID\":22222,\"Gecos\":\"User2\",\"Dir\":\"/home/user2\",\"Shell\":\"/bin/dash\",\"LastPwdChange\":-1,\"MaxPwdAge\":-1,\"PwdWarnPeriod\":-1,\"PwdInactivity\":-1,\"MinPwdAge\":-1,\"ExpirationDate\":-1,\"LastLogin\":\"BBBBBTIME\"}'\n    \"3333\": '{\"Name\":\"user3\",\"UID\":3333,\"GID\":33333,\"Gecos\":\"User3\",\"Dir\":\"/home/user3\",\"Shell\":\"/bin/zsh\",\"LastPwdChange\":-1,\"MaxPwdAge\":-1,\"PwdWarnPeriod\":-1,\"PwdInactivity\":-1,\"MinPwdAge\":-1,\"ExpirationDate\":-1,\"LastLogin\":\"ABCDETIME\"}'\nUserByName:\n    user1: '{\"Name\":\"user1\",\"UID\":1111,\"GID\":11111,\"Gecos\":\"User1 gecos\\nOn multiple lines\",\"Dir\":\"/home/user1\",\"Shell\":\"/bin/bash\",\"LastPwdChange\":-1,\"MaxPwdAge\":-1,\"PwdWarnPeriod\":-1,\"PwdInactivity\":-1,\"MinPwdAge\":-1,\"ExpirationDate\":-1,\"LastLogin\":\"AAAAATIME\"}'\n    user2: '{\"Name\":\"user2\",\"UID\":2222,\"GID\":22222,\"Gecos\":\"User2\",\"Dir\":\"/home/user2\",\"Shell\":\"/bin/dash\",\"LastPwdChange\":-1,\"MaxPwdAge\":-1,\"PwdWarnPeriod\":-1,\"PwdInactivity\":-1,\"MinPwdAge\":-1,\"ExpirationDate\":-1,\"LastLogin\":\"BBBBBTIME\"}'\n    user3: '{\"Name\":\"user3\",\"UID\":3333,\"GID\":33333,\"Gecos\":\"User3\",\"Dir\":\"/home/user3\",\"Shell\":\"/bin/zsh\",\"LastPwdChange\":-1,\"MaxPwdAge\":-1,\"PwdWarnPeriod\":-1,\"PwdInactivity\":-1,\"MinPwdAge\":-1,\"ExpirationDate\":-1,\"LastLogin\":\"ABCDETIME\"}'\nUserToGroups:\n    \"1111\": '{\"UID\":1111,\"GIDs\":[11111]}'\n    \"2222\": '{\"UID\":2222,\"GIDs\":[22222,99999]}'\n    \"3333\": '{\"UID\":3333,\"GIDs\":[33333,99999]}'\n"
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,2 +1,4 @@
                                 GroupByID:
                                +    "11111": '{"Name":"group1","GID":11111}'
                                +    "22222": '{"Name":"group2","GID":22222}'
                                     "33333": '{"Name":"group3","GID":33333}'
                                @@ -5,11 +7,21 @@
                                     commongroup: '{"Name":"commongroup","GID":99999}'
                                +    group1: '{"Name":"group1","GID":11111}'
                                +    group2: '{"Name":"group2","GID":22222}'
                                     group3: '{"Name":"group3","GID":33333}'
                                 GroupToUsers:
                                +    "11111": '{"GID":11111,"UIDs":[1111]}'
                                +    "22222": '{"GID":22222,"UIDs":[2222]}'
                                     "33333": '{"GID":33333,"UIDs":[3333]}'
                                -    "99999": '{"GID":99999,"UIDs":[3333]}'
                                +    "99999": '{"GID":99999,"UIDs":[2222,3333]}'
                                 UserByID:
                                +    "1111": '{"Name":"user1","UID":1111,"GID":11111,"Gecos":"User1 gecos\nOn multiple lines","Dir":"/home/user1","Shell":"/bin/bash","LastPwdChange":-1,"MaxPwdAge":-1,"PwdWarnPeriod":-1,"PwdInactivity":-1,"MinPwdAge":-1,"ExpirationDate":-1,"LastLogin":"AAAAATIME"}'
                                +    "2222": '{"Name":"user2","UID":2222,"GID":22222,"Gecos":"User2","Dir":"/home/user2","Shell":"/bin/dash","LastPwdChange":-1,"MaxPwdAge":-1,"PwdWarnPeriod":-1,"PwdInactivity":-1,"MinPwdAge":-1,"ExpirationDate":-1,"LastLogin":"BBBBBTIME"}'
                                     "3333": '{"Name":"user3","UID":3333,"GID":33333,"Gecos":"User3","Dir":"/home/user3","Shell":"/bin/zsh","LastPwdChange":-1,"MaxPwdAge":-1,"PwdWarnPeriod":-1,"PwdInactivity":-1,"MinPwdAge":-1,"ExpirationDate":-1,"LastLogin":"ABCDETIME"}'
                                 UserByName:
                                +    user1: '{"Name":"user1","UID":1111,"GID":11111,"Gecos":"User1 gecos\nOn multiple lines","Dir":"/home/user1","Shell":"/bin/bash","LastPwdChange":-1,"MaxPwdAge":-1,"PwdWarnPeriod":-1,"PwdInactivity":-1,"MinPwdAge":-1,"ExpirationDate":-1,"LastLogin":"AAAAATIME"}'
                                +    user2: '{"Name":"user2","UID":2222,"GID":22222,"Gecos":"User2","Dir":"/home/user2","Shell":"/bin/dash","LastPwdChange":-1,"MaxPwdAge":-1,"PwdWarnPeriod":-1,"PwdInactivity":-1,"MinPwdAge":-1,"ExpirationDate":-1,"LastLogin":"BBBBBTIME"}'
                                     user3: '{"Name":"user3","UID":3333,"GID":33333,"Gecos":"User3","Dir":"/home/user3","Shell":"/bin/zsh","LastPwdChange":-1,"MaxPwdAge":-1,"PwdWarnPeriod":-1,"PwdInactivity":-1,"MinPwdAge":-1,"ExpirationDate":-1,"LastLogin":"ABCDETIME"}'
                                 UserToGroups:
                                +    "1111": '{"UID":1111,"GIDs":[11111]}'
                                +    "2222": '{"UID":2222,"GIDs":[22222,99999]}'
                                     "3333": '{"UID":3333,"GIDs":[33333,99999]}'
                Test:           TestNew/Clean_up_on_interval
                Messages:       Did not get expected database content
FAIL
FAIL    github.com/ubuntu/authd/internal/cache  2.199s
```

Probably rare, but worth checking what could be racy!

### Steps to reproduce it

Run tests

### Ubuntu users: System information and logs

_No response_

### Non Ubuntu users: System information and logs

_No response_

### Relevant information

_No response_

### Double check your logs

- [X] I have redacted any sensitive information from the logs

--- a/internal/cache/db.go
+++ b/internal/cache/db.go
@@ -342,6 +342,10 @@ func getActiveUsers(procDir string) (activeUsers map[string]struct{}, err error)
 
 		info, err := dirEntry.Info()
 		if err != nil {
+			// If the file doesn't exist, it means the process is not running anymore so we can ignore it.
+			if errors.Is(err, fs.ErrNotExist) {
+				continue
+			}
 			return nil, err
 		}
 

